src/types.rs

```diff
@@ -30,9 +30,389 @@ pub struct DisplayName;
 pub struct Nickname;
 
 /// RFC3339 timestamp
-#[aliri_braid::braid(serde)]
+#[aliri_braid::braid(serde, validator)]
 pub struct Timestamp;
 
+impl aliri_braid::Validator for Timestamp {
+    type Error = TimestampParseError;
+
+    fn validate(s: &str) -> Result<(), Self::Error> {
+        #[cfg(feature = "chrono")]
+        {
+            let _ = chrono::DateTime::<chrono::FixedOffset>::parse_from_rfc3339(s)?;
+            Ok(())
+        }
+        #[cfg(not(feature = "chrono"))]
+        {
+            // This validator is lacking some features for now
+            if !s.chars().all(|c| {
+                c.is_numeric()
+                    || c == 'T'
+                    || c == 'Z'
+                    || c == '+'
+                    || c == '.'
+                    || c == '-'
+                    || c == ':'
+            }) {
+                return Err(TimestampParseError::invalid());
+            }
+            // PSA: Don't do time stuff... it sucks
+            if let Some(i) = s.find('T') {
+                // if no `T`, then it's not a valid timestamp
+                if i < 1 {
+                    return Err(TimestampParseError::invalid());
+                };
+                let (full_date, full_time) = s.split_at(i);
+                if full_date.len() != "1900-00-00".len() {
+                    return Err(TimestampParseError::invalid_s(full_date));
+                }
+                if !full_date.chars().all(|c| c.is_numeric() || c == '-') {
+                    return Err(TimestampParseError::invalid_s(full_date));
+                }
+                let partial_time = if let Some(stripped) = full_time.strip_suffix('Z') {
+                    stripped
+                } else {
+                    // time is offset from UTC.
+                    &full_time[..full_time.len() - "+00:00".len()]
+                };
+                if 2 != partial_time
+                    .chars()
+                    .into_iter()
+                    .filter(|&b| b == ':')
+                    .count()
+                {
+                    return Err(TimestampParseError::invalid_s(partial_time));
+                };
+                if !partial_time.contains('.') && partial_time.len() != "T00:00:00".len() {
+                    return Err(TimestampParseError::invalid_s(partial_time));
+                } else if partial_time.contains('.') {
+                    let mut i = partial_time.split('.');
+                    // if len not correct or next is none
+                    if !i
+                        .next()
+                        .map(|s| s.len() == "T00:00:00".len())
+                        .unwrap_or_default()
+                    {
+                        return Err(TimestampParseError::invalid_s(partial_time));
+                    }
+                }
+            } else {
+                return Err(TimestampParseError::invalid());
+            }
+
+            Ok(())
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
```

Comment:
this test isn't really needed as all the other tests using `Timestamp` already define pretty well what a timestamp is.

Codecov for this would be nice though